__tests__/test_ui_components/test_utils.js

```diff
@@ -0,0 +1,72 @@
+import { Provider } from "react-redux";
+import { mount, shallow, render } from "enzyme";
+import { createStore } from "redux";
+import renderer from "react-test-renderer";
+import reducers from "../../src/js/reducers";
+import { initData } from "../../src/js/actions/";
+import assetData from "../../src/assets/assets.json";
+import {checkPropsExist, deepCopy} from "../../src/js/utils.js";
+
+const testProps = [
+	"store",
+	"qset",
+	"currentX",
+	"currentZ"
+];
+
+const assets = {
+	bed: {
+		id: "bed-1",
+		category: "beds",
+		objSrc: "url/to/obj",
+		mtlSrc: "url/to/mtl"
+	},
+	chair: {
+		id: "chair",
+		category: "equipment",
+		objSrc: "url/to/obj",
+		mtlSrc: "url/to/mtl"
+	},
+	wall: {
+		id: "chair",
+		category: "equipment"
+	}
+};
+
+let deepCopyMock = jest.fn();
+
+describe("Utils tests", () => {
+
+	it("should console log if caller exists in checkPropsExist", () => { 
+		const consoleSpy = jest.spyOn(console, 'log').mockReturnValue("test");
+		checkPropsExist(testProps, true);
+		expect(consoleSpy).toBeCalled();
+	});
+
+	it("should not console log if caller is false in checkPropsExist", () => { 
+		const consoleSpy = jest.spyOn(console, 'log').mockReturnValue("test");
+		checkPropsExist(testProps, false);
+		expect(consoleSpy).toBeCalled();
+	});
+
+	it("should deepCopy an object", () => { 
+		let deepCopyFunction = deepCopy(assets);
+		deepCopyMock();
```

Comment:
These 3 tests appear to only be expecting a mock function to return - not really looking at the  results of deepCopy.  This will get the coverage up, but the expect tests aren't really validating anything.

Suggest making sure the returned object looks like assets, but doesn't share any of the actual objects from the original.

Reply:
Changed the tests to actually test whether or not deepCopy is working.